pytorch_lightning/loops/closure.py

```diff
@@ -53,25 +54,29 @@ def __init__(self) -> None:
         super().__init__()
         self._result: Optional[ClosureResult] = None
 
-    def get_result(self) -> Optional[ClosureResult]:
+    @property
+    def result(self) -> ClosureResult:
         """The cached result from the last time the closure was called.
 
         Once accessed, the internal reference gets reset and the consumer will have to hold on to the reference as long
         as necessary.
         """
-        result = self._result
-        self._result = None  # free memory
+        if self._result is None:
+            raise MisconfigurationException(
+                "The closure hasn't been executed."
+                " HINT: did you call `optimizer_closure()` in your `optimizer_step` hook?"
```

Comment:
maybe the hint should suggest that the optimizer calls the closure? If an optimizer didn't do this, then an error like this would also be very helpful right?